src/WebJobs.Extensions.OpenAI/Search/ISearchProvider.cs

```diff
@@ -10,6 +10,8 @@ namespace Microsoft.Azure.WebJobs.Extensions.OpenAI.Search;
 /// </summary>
 public interface ISearchProvider
 {
+    string Name { get; set; }
```

Comment:
Let's add `///` documentation comments since this is a public API.

Also, this should just have a getter and not a setter.

```suggestion
    string Name { get; }
```